instrumentation/opentelemetry-instrumentation-urllib/tests/test_metrics_instrumentation.py

```diff
@@ -0,0 +1,248 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import urllib
+from urllib import request
+from urllib.parse import urlencode
+from timeit import default_timer
+from typing import Optional, Union
+
+import httpretty
+from opentelemetry.semconv.metrics import MetricInstruments
+
+from opentelemetry.instrumentation.urllib import (  # pylint: disable=no-name-in-module,import-error
+    URLLibInstrumentor,
+)
+from opentelemetry.sdk.metrics._internal.point import Metric
+from opentelemetry.sdk.metrics.export import (
+    HistogramDataPoint,
+    NumberDataPoint,
+)
+from opentelemetry.test.test_base import TestBase
+
+
+class TestRequestsIntegration(TestBase):
+    URL = "http://httpbin.org/status/200"
+    URL_POST = "http://httpbin.org/post"
+
+    def setUp(self):
+        super().setUp()
+        URLLibInstrumentor().instrument()
+        httpretty.enable()
+        httpretty.register_uri(httpretty.GET, self.URL, body=b"Hello!")
+        httpretty.register_uri(
+            httpretty.POST, self.URL_POST, body=b"Hello World!"
+        )
+
+    def tearDown(self):
+        super().tearDown()
+        URLLibInstrumentor().uninstrument()
+        httpretty.disable()
+
+    def get_sorted_metrics(self):
+        resource_metrics = (
+            self.memory_metrics_reader.get_metrics_data().resource_metrics
+        )
+
+        all_metrics = []
+        for metrics in resource_metrics:
+            for scope_metrics in metrics.scope_metrics:
+                all_metrics.extend(scope_metrics.metrics)
+
+        return self.sorted_metrics(all_metrics)
+
+    @staticmethod
+    def sorted_metrics(metrics):
+        """
+        Sorts metrics by metric name.
+        """
+        return sorted(
+            metrics,
+            key=lambda m: m.name,
+        )
+
+    def assert_metric_expected(
```

Comment:
I used utils function for metrics assert
they can be deleted after https://github.com/open-telemetry/opentelemetry-python/pull/3092 merged

Reply:
I have some reservations about the PR on the main repo. I will leave a review there about it.

Reply:
Thank you!